### PR TITLE
storage: add memory quota metrics

### PR DESCRIPTION
**Pull request by overvenus**
### What is changed and how it works?

Issue Number: ref #16234 

What's Changed:

```commit-message
* Add a metric of scheduler memory quota.
* Add a metric of scheduler running commands.
```

### Check List

Tests 

- [x] Manual test (add detailed scripts or steps below)

### Release note

```release-note
None
```

### CODE REVIEW

File: src/storage/txn/scheduler.rs [hunk 1]
```diff
@@ -495,6 +499,15 @@ impl<E: Engine, L: LockManager> TxnScheduler<E, L> {
         self.inner.scale_pool_size(pool_size)
     }
 
+    pub fn memory_quota_capacity(&self) -> usize {
+        self.inner.memory_quota.capacity()
+    }
+
+    pub(in crate::storage) fn set_memory_quota_capacity(&self, cap: usize) {
+        SCHED_TXN_MEMORY_QUOTA.allocated.set(cap as i64);
```
Comment (glorv) [inline src/storage/txn/scheduler.rs:507]: The label "allocated" actually means the capacity of the total memory can be used by scheduler tasks, so is it better to rename it with "capacity"? The memory is not pre-allocated, the value of "used" should be the amount of allocated memory

**Review by glorv — APPROVED** [review]
LGTM, please remember to update the online config document.

File: src/storage/metrics.rs [hunk 2]
```diff
@@ -617,4 +624,18 @@ lazy_static! {
...
+
+    pub static ref SCHED_TXN_MEMORY_QUOTA: MemoryQuotaGauge = register_static_int_gauge_vec!(
+        MemoryQuotaGauge,
+        "tikv_scheduler_memory_quota_size",
+        "Statistics of in_use and capacity of scheduler memory quota",
+        &["type"]
+    )
+    .unwrap();
+
+    pub static ref SCHED_TXN_RUNNING_COMMANDS: IntGauge = register_int_gauge!(
```
Comment (cfzjywxk) [inline src/storage/metrics.rs:636]: Currently, there is a pending command statistic that is more closely related to memory usage at https://github.com/tikv/tikv/pull/16482/files#diff-fd86c97f137c667fa1222b07fd71d8e369b7ada70a34afc8415d1665ef9ea9e8L407. Is it still necessary to have `running`?

**Review by Connor1996 — APPROVED** [review]
LGTM

File: src/storage/metrics.rs (hunk 2 above)
Comment (overvenus) [reply src/storage/metrics.rs:636]: The future spawns by a `running` txn command takes about 6184 bytes, I think it's worth to record, also it reflects how busy the scheduler pool is.

**Review by cfzjywxk — APPROVED** [review]
LGTM

**Pull request merged**